Add plugin `Supsub`

wjgoarxiv opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/wjgoarxiv/obsidian-supsub

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["description": "`Supsub` is a plugin designed for Obsidian.md that adds functionality to wrap selected text with](https://github.com/wjgoarxiv/obsidian-supsub/blob/ba63a5756c871d9be2f2ae638cefc10910d46b7e/manifest.json#L6)
We have recently published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.
Avoid including sentences like "This is a plugin for Obsidian" in your description.

Where is your source code?
Or have you made all your changes in the `main.js` file directly?


[hotkeys: [](https://github.com/wjgoarxiv/obsidian-supsub/blob/ba63a5756c871d9be2f2ae638cefc10910d46b7e/main.js#L38-L43)
While there's no hard rule against it, we recommend against providing a default hotkey when possible. The reason being that it's easy to pick a hotkey that a user already has configured and they could get confused when that key doesn't do what they expect. Also, it's hard choosing a safe default hotkey that's available for all operating systems.

[const activeLeaf = this.app.workspace.activeLeaf;](https://github.com/wjgoarxiv/obsidian-supsub/blob/ba63a5756c871d9be2f2ae638cefc10910d46b7e/main.js#L59-L63)
`activeLeaf` is deprecated and should not be used.

[const editor = activeLeaf.view.sourceMode.cmEditor;](https://github.com/wjgoarxiv/obsidian-supsub/blob/ba63a5756c871d9be2f2ae638cefc10910d46b7e/main.js#L62)
`cmEditor` is only for the legacy editor, and should not be used.

Instead of a simple `callback` for your commands you should use a `editorCallback` instead, this will solve the two issues above.
https://docs.obsidian.md/Plugins/User+interface/Commands#Editor+commands

wjgoarxiv commented:
**Where is your source code?**
→ Yes, I've made all my changes in the `main.js` file directly. 

**`activeLeaf` is deprecated and should not be used.** && **`cmEditor` is only for the legacy editor, and should not be used.**
→ I revised the script; I removed the hotkeys and switched from using `activeLeaf` and `cmEditor` to using the editor passed into the `editorCallback` function. May work with both the legacy and the new editor in Obsidian.

joethei commented:
Editing the compiled `main.js` directly is not recommended.
Please either use TypeScript following our template: <https://github.com/obsidianmd/obsidian-sample-plugin>, or update your code to use JavaScript properly.
Currently the first ~30 lines of your code are not needed.
